resources/prosody-plugins/mod_auth_token.lua

```diff
@@ -74,18 +74,28 @@ function provider.delete_user(username)
 	return nil;
 end
 
+local function validate_result(session, res, error, reason)
+    if res == false then
+        log("warn",
+            "Error verifying token err:%s, reason:%s", error, reason);
+        session.auth_token = nil;
+        return res, error, reason;
+    end
+end
+
 function provider.get_sasl_handler(session)
 
 	local function get_username_from_token(self, message)
-        local res, error, reason = token_util:process_and_verify_token(session);
 
-        if (res == false) then
-            log("warn",
-                "Error verifying token err:%s, reason:%s", error, reason);
-            session.auth_token = nil;
-            return res, error, reason;
+        -- retrieve custom public key from server and save it on the session
+        local event_result = prosody.events.fire_event("pre-jitsi-authentication-fetch-key", session);
+        if event_result ~= nil then
+            validate_result(session,event_result.res, event_result.error, event_result.reason)
```

Comment:
We aren't returning anything from get_username_from_token, so this will never reject bad users.  I don't think we need this function, but if we decide we do, we should at least say: `return validate_result` rather than just running the function.

Reply:
I remove it, sorry for the subtle bug that I introduced with this function.